### PR TITLE
feat: card api integration

### PR DESCRIPTION
**Pull request by bharathraj25**
* fixed some spelling mistakes
* fixed: made due date optional
* refactored some functions
* integrated api on cards

### CODE REVIEW

File: src/components/contexts/ErrorContext.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  return (
+    <ErrorContext.Provider value={errorContextValues}>
+      {children}
+      <Box>
+        {error !== '' && (<Box><ErrorSnackbar message={error} setError={setError} /></Box>)}
+        {success !== '' && (<Box><SuccessSnackbar message={success} setSuccess={setSuccess} /></Box>)}
+      </Box>
+    </ErrorContext.Provider>
+  );
+
```
Comment (sugumar-mani) [inline src/components/contexts/ErrorContext.js:24]: unnecessary empty line

File: src/components/utilityFunctions/DateGetter.js [hunk 2]
```diff
@@ -1,13 +1,13 @@
-
-const dateGetter = (timeStamp,dateHolder) => {
+const dateGetter = (timeStamp, dateHolder) => {
+    if (timeStamp === null) return "[DD Mon, YYYY]";
```
Comment (sugumar-mani) [inline src/components/utilityFunctions/DateGetter.js:2]: declare constant for raw string

File: src/components/poNotesComponents/poNotesTables/PONotesGridLayout.js [hunk 3]
```diff
@@ -1,52 +1,82 @@
...
+const keyDecisions = {
+  heading: 'Key Decisions',
+  definition: ' are the vital outcomes/decisions from the various discussions that PO has been part of.',
+  accessibilityInformation: '  PO is the owner of this section only PO can add or edit these entries.'
+}
 
 export default function PONotesGridLayout() {
+  // setting serach query based on { text, start date , end date and status }.
+  const query = {
+    // search: "a",
```
Comment (sugumar-mani) [inline src/components/poNotesComponents/poNotesTables/PONotesGridLayout.js:30]: do we nee these?

File: src/components/poNotesComponents/PONotesBody.js [hunk 4]
```diff
@@ -1,9 +1,8 @@
 import React from 'react';
 import Grid from '@mui/material/Grid';
-
 import PONotesGridLayout from './poNotesTables/PONotesGridLayout';
 
-export default function poNotesBody() {
+export default function PoNotesBody() {
```
Comment (sugumar-mani) [inline src/components/poNotesComponents/PONotesBody.js:5]: i would prefer PO istead Po. either bot caps or both small case

File: src/components/poNotesComponents/poNotesTables/PONotesGridLayout.js (hunk 3 above)
Comment (bharathraj25) [reply src/components/poNotesComponents/poNotesTables/PONotesGridLayout.js:30]: No, it was used for testing out query.
Will remove this

File: src/components/poNotesComponents/PONotesBody.js (hunk 4 above)
Comment (bharathraj25) [reply src/components/poNotesComponents/PONotesBody.js:5]: Makes sense, will correct those

**Review by sugumar-mani — APPROVED** [review]

File: src/components/contexts/ErrorContext.js (hunk 1 above)
Comment (bharathraj25) [reply src/components/contexts/ErrorContext.js]: removed it

File: src/components/utilityFunctions/DateGetter.js (hunk 2 above)
Comment (bharathraj25) [reply src/components/utilityFunctions/DateGetter.js]: fixed it

**Pull request merged**